### DISCUSSION TITLE
fvSource  field optimization

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#623] [General]
Hi,

I have a rather conceptual question. 

In the DASimpleFOAM solver, we have this in the UEqn.H

```
tmp<fvVectorMatrix> tUEqn(
    fvm::div(phi, U, divUScheme)
    + fvm::Sp(alphaPorosity, U)
    + MRF.DDt(U)
    + turbulencePtr_->divDevReff(U)
    - fvSource
    - fvOptions(U));
```

If my understanding is correct, this creates a matrix with co-efficients using each one of those terms (since all of them are implicit).

I am not able to interpret this fvSource term. Is it a body force? Is it implicit or explicit? Is it related to the velocity? 

We have obtained some results for the distribution of fvSource and it is meaningful but we want to use it for some applications and hence we need to interpret it. 

Please advice!

**Comment by friedenhe** [comment]
The fvSource is a body force, and it is treated explicitly. It is not related to the velocity.
